pkg/zsio/json/parser.go

```diff
@@ -0,0 +1,145 @@
+package json
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/buger/jsonparser"
+	"github.com/mccanne/zq/pkg/zeek"
+	"github.com/mccanne/zq/pkg/zson"
+	"github.com/mccanne/zq/pkg/zval"
+)
+
+// NewRawFromJSON returns a new zson.Raw slice as well as an inferred zeek.Type
+// from provided block of JSON. The function expects the input json to be an
+// object, otherwise an error is returned.
+func NewRawFromJSON(b []byte) (zson.Raw, zeek.Type, error) {
+	val, typ, _, err := jsonparser.Get(b)
+	if err != nil {
+		return nil, nil, err
+	}
+	if typ != jsonparser.Object {
+		return nil, nil, fmt.Errorf("expected json type to be Object got %v", typ)
```

Comment:
Nits: s/json/JSON/  and either s/Object got/Object but got/ or s/Object got/Object, got/.

Maybe also s/%v/%#v/ since %#v will include the Go type.